cardinal_toe_app/src/app/services/map-generator/map-generator.service.ts

```diff
@@ -0,0 +1,163 @@
+import { Injectable, OnInit } from '@angular/core';
+import { Location } from '../../interfaces/Location';
+import { LocationGeneratorService } from '../location-generator/location-generator.service';
+import { NumberRandomiserService } from '../number-randomiser/number-randomiser.service';
+import locationsJSON from '../../../assets/data/locations.json';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MapGeneratorService {
+  
+  // Declare constants
+  /**
+   * Sets the limit of the number of nodes (or, distance) that can be generated in the node tree
+   * away from the _'winning route'_.
+   * @private
+   * @readonly
+   * @constant
+   */
+  private readonly DEVIATION_LIMIT = 3;
+    /**
+   * Sets the limit of the number of Locations (i.e. distance) that can be generated in the Location tree
+   * before reaching the _'goal Location'_.
+   * @private
+   * @readonly
+   * @constant
+   */
+  private readonly DEPTH_LIMIT = 10;
+
+  rootLocation!: Location;
+
+  constructor ( 
+    private locationGeneratorService: LocationGeneratorService, 
+    private numberRandomiserService: NumberRandomiserService 
+  ) { }
+
+
+  getMap(): Location {
+    console.log(`Getting map...`)
+    this.generateMap();
+    return this.rootLocation;
+  }
+
+
+  /**
+   * ???
+   * @param depthValue      Tracks the number steps back to `rootLocation`.
+   * @param deviationValue  Tracks number of steps back to _'winning route'_.
+   * @param currentLocation ???
+   * @param isCorrectRoute  OPTIONAL: tracks if `currentLocation` leads to `isFinalLocation` (i.e. is on the _'winning route'_).
+   */
+  generateMap (
+      depthValue: number = 0, 
+      deviationValue: number = 0,
+      currentLocation: Location | null = null, 
+    ) {
+
+      
+
```

Comment:
I wouldn't start a function with this much whitespace. In general I'd only add a line to break things up and never at the start of a function.